include/ctarget_set.h

```diff
@@ -0,0 +1,199 @@
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+ * @section DESCRIPTION
+ * Target-specific runtime functions for the C target language.
+ * This API layer can be used in conjunction with:
+ *     target C;
+ * 
+ * Note for target language developers. This is one way of developing a target language where 
+ * the C core runtime is adopted. This file is a translation layer that implements Lingua Franca 
+ * APIs which interact with the internal _lf_SET and _lf_schedule APIs. This file can act as a 
+ * template for future runtime developement for target languages.
+ * For source generation, see xtext/org.icyphy.linguafranca/src/org/icyphy/generator/CCppGenerator.xtend.
+ */
+
+#ifndef CTARGET_SET
+#define CTARGET_SET
+
+//////////////////////////////////////////////////////////////
+/////////////  SET Functions (to produce an output)
+
+// NOTE: According to the "Swallowing the Semicolon" section on this page:
+//    https://gcc.gnu.org/onlinedocs/gcc-3.0.1/cpp_3.html
+// the following macros should use an odd do-while construct to avoid
+// problems with if ... else statements that do not use braces around the
+// two branches.
+
+/**
+ * Set the specified output (or input of a contained reactor)
+ * to the specified value.
+ *
+ * If the value argument is a primitive type such as int,
+ * double, etc. as well as the built-in types bool and string, 
+ * the value is copied and therefore the variable carrying the
+ * value can be subsequently modified without changing the output.
+ * This also applies to structs with a type defined by a typedef
+ * so that the type designating string does not end in '*'.
+ * 
+ * If the value argument is a pointer
+ * to memory that the calling reaction has dynamically allocated,
+ * the memory will be automatically freed once all downstream
+ * reactions no longer need the value.
+ * If 'lf_set_destructor' is called on 'out', then that destructor
+ * will be used to free 'value'. 
+ * Otherwise, the default void free(void*) function is used.
+ * 
+ * @param out The output port (by name) or input of a contained
+ *  reactor in form input_name.port_name.
+ * @param value The value to insert into the self struct.
+ */
+#define lf_set(out, val) _LF_SET(out, val)
+#define SET(out, val) \
```

Comment:
Is a `do-while` needed here?

See https://gcc.gnu.org/onlinedocs/cpp/Swallowing-the-Semicolon.html.